include/boost/geometry/algorithms/detail/disjoint/multirange_geometry.hpp

```diff
@@ -68,13 +68,12 @@ struct multirange_constant_size_geometry
                         typename boost::range_value<MultiRange>::type,
                         ConstantSizeGeometry
                     >
-            > unary_predicate_type;
+            >;
 
-        return detail::check_iterator_range
-            <
-                unary_predicate_type
-            >::apply(boost::begin(multirange), boost::end(multirange),
-                     unary_predicate_type(constant_size_geometry, strategy));
+        return boost::end(multirange) == 
+            std::find_if(boost::begin(multirange),
+                         boost::end(multirange), 
+                         unary_predicate_type(constant_size_geometry, strategy));
```

Comment:
Find if a point/segment/box intersects a geometry in a range and if not then return true. For empty range the result is true as well, ok.